packages/pymedphys_dicom/src/pymedphys_dicom/dicom/_level2/anonymise.py

```diff
@@ -373,39 +375,111 @@ def anonymise_directory(
         information.
     """
     dicom_filepaths = glob(dicom_dirpath + '/**/*.dcm', recursive=True)
+    failing_filepaths = []
+    errors = []
 
     for dicom_filepath in dicom_filepaths:
-        anonymise_file(
-            dicom_filepath,
-            delete_original_file=False,
-            anonymise_filename=anonymise_filenames,
-            replace_values=replace_values,
-            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
-            delete_private_tags=delete_private_tags,
-            delete_unknown_tags=delete_unknown_tags)
+        try:
+            anonymise_file(
+                dicom_filepath,
+                delete_original_file=False,
+                anonymise_filename=anonymise_filenames,
+                replace_values=replace_values,
+                keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+                delete_private_tags=delete_private_tags,
+                delete_unknown_tags=delete_unknown_tags)
+        except Exception as e:
+            failing_filepaths.append(dicom_filepath)
+            errors.append(e)
 
     # Separate loop provides the ability to raise Exceptions from the
     # unsuccessful deletion of the original DICOM files while preventing
     # these Exceptions from interrupting the batch anonymisation.
```

Comment:
This is quite a neat way to handle it.

Reply:
If there are any errors though I do wonder if it would be better simply to "fail fast" and not go any further. Let users sort out what's wrong before running over the rest of the directory...